src/models/Factor.js

```diff
@@ -235,9 +240,17 @@ function (Okta, Q, factorUtil, BaseLoginModel) {
 
         return promise
         .then(function (trans) {
+          var options = {
+            'delay': PUSH_INTERVAL
+          };
           if (trans.status === 'MFA_CHALLENGE' && trans.poll) {
             return Q.delay(PUSH_INTERVAL).then(function() {
-              return trans.poll(PUSH_INTERVAL);
+              if (self.settings.get('features.autoPush') && self.get('factorType') === 'push') {
+                options.autoPush = function() {
```

Comment:
Should this be a fn here?

Reply:
Yea we are implementing this as a function so if the user changes the checkbox value during an active poll, it will catch that change and update the backend. - https://github.com/okta/okta-auth-js/pull/91/files#diff-1a136b779b30ddb4520bd33f89a5c8afR81